quickwit/quickwit-search/src/leaf.rs

```diff
@@ -402,21 +407,25 @@ pub async fn leaf_search(
         Vec<(String, SearchError)>,
     ) = split_search_results
         .into_iter()
-        .partition_map(|split_search_res| match split_search_res {
-            Ok(split_search_resp) => Either::Left(Ok(split_search_resp)),
-            Err(err) => Either::Right(err),
-        });
+        .partition_map(
+            |split_search_res| match split_search_res.expect("spawned future died") {
```

Comment:
I understand that this is the same behavior as before, but could we make that expect into a SearchError sayin the taks pannicked or was cancelled?